docs/Migration-Guide-V4.md

```diff
@@ -0,0 +1,12 @@
+# V4 Migration Guide
+
+This guide is intended to help with migration from Fastify v3 to v4.
+
+Before migrating to v4, please ensure that you have fixed all deprecation warningx from v3.
+All v3 deprecations have been removed and they will no longer work after upgrading.
+
+## Breaking Changes
+
+### Deprecation of `app.use()`
+
+Starting this version of Fastify, we have deprecated the use of `app.use()`. We have decided not support the use of middlewares. Both `middie` and `fastify-express` will still be there and maintained.
```

Comment:
We should suggest to users what use instead of middlewares

```suggestion
Starting this version of Fastify, we have deprecated the use of `app.use()`. We have decided not to support the use of middlewares. Both `middie` and `fastify-express` will still be there and maintained. Use Fastify's hooks instead.
```

Reply:
Agreed!